core/src/main/scala/org/apache/spark/storage/StorageStatusListener.scala

```diff
@@ -87,8 +94,13 @@ class StorageStatusListener extends SparkListener {
   override def onBlockManagerRemoved(blockManagerRemoved: SparkListenerBlockManagerRemoved) {
     synchronized {
       val executorId = blockManagerRemoved.blockManagerId.executorId
-      executorIdToStorageStatus.remove(executorId)
+      val removedStorageStatus = executorIdToStorageStatus.remove(executorId)
```

Comment:
nit: looks nicer if you do `executorIdToStorageStatus.remove(executorId).foreach { status =>`